src/mlpack/methods/ann/layer/batch_norm.hpp

```diff
@@ -44,39 +47,65 @@ namespace ann /** Artificial Neural Network. */ {
  * }
  * @endcode
  *
- * @tparam InputDataType Type of the input data (arma::colvec, arma::mat,
- *         arma::sp_mat or arma::cube).
- * @tparam OutputDataType Type of the output data (arma::colvec, arma::mat,
- *         arma::sp_mat or arma::cube).
+ * @tparam MatType Matrix representation to accept as input and use for
+ *         computation.
  */
-template <
-  typename InputType = arma::mat,
-  typename OutputType = arma::mat
->
-class BatchNorm : public Layer<InputType, OutputType>
+template <typename MatType = arma::mat>
+class BatchNormType : public Layer<MatType>
 {
  public:
-  //! Create the BatchNorm object.
-  BatchNorm();
+  //! Create the BatchNorm object. This will initialize maxAxis to 2.
+  //! (or effectively 1 or 0 if there are fewer input Dimensions than 3.)
+  BatchNormType();
 
   /**
    * Create the BatchNorm layer object for a specified number of input units.
    *
```

Comment:
```suggestion
   * Create the BatchNorm layer object for a specified number of input units.
   * The `maxAxis` parameter sets the maximum axis to apply batch
   * normalization to; dimensions higher than this dimension are left
   * untouched.  As an example, if we have a 3-dimensional input (call the
   * three dimensions rows, columns, and slices), and `maxAxis` is 2, then we
   * apply the same normalization across different slices.
```
I adapted the comment above for here too.